File: app/src/release/java/com/handstandsam/shoppingapp/MyApplication.java
package com.handstandsam.shoppingapp;

import com.handstandsam.shoppingapp.di.AppComponent;
import com.handstandsam.shoppingapp.di.AppModule;
import com.handstandsam.shoppingapp.di.DaggerAppComponent;
import com.handstandsam.shoppingapp.di.NetworkModule;
import com.handstandsam.shoppingapp.di.RepositoryModule;

public class MyApplication extends MyAbstractApplication {

    @Override
    protected AppComponent createAppComponent() {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .networkModule(new NetworkModule(NetworkModule.LOCALHOST_ENDPOINT))
                .repositoryModule(new RepositoryModule(this))
                .build();
    }

}

File: app/src/release/java/com/handstandsam/shoppingapp/MyApplication.kt
package com.handstandsam.shoppingapp

import com.handstandsam.shoppingapp.di.*

class MyApplication : MyAbstractApplication() {

    override fun createAppComponent(endpoint: String): AppComponent {
        return DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .networkModule(NetworkModule(NetworkModule.LOCALHOST_ENDPOINT))
                .repositoryModule(RepositoryModule(this))
                .build()
    }
}
